### PR TITLE
feat(core): allow configuration of default button options

### LINKED ISSUE #133: 🚀 - Use providers to inject default configurations
# 🚀 Feature request

### Is your feature request related to a problem?

I would like to adjust the default values for several inputs on different components. In our project for instance, we would like to use the checkbox component in size `l`. Given the current implementation, we need to add the size input to every checkbox. It would be nice to globally define the default values via a configuration provider. Similar functionality exists on Angular Material and other frameworks.

### Describe the solution you'd like

```
import {TUI_CHECKBOX_DEFAULT_OPTIONS} from '@taiga-ui/kit';
...
{
  imports: [TuiRootModule, ...],
  providers: [{
    provide: TUI_CHECKBOX_DEFAULT_OPTIONS,
    useValue: {
      size: 'l'
    }
  }],
  ...
}
export class AppModule {}
```

### Describe alternatives you've considered

The only alternative currently is to set the size input on every checkbox.

### Additional context

Ideally, this would be available for every component and not only for checkboxes. We can help provide PRs if you like the idea.

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [ ] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Refactoring
-   [ ] Code style update
-   [ ] Build or CI related changes
-   [ ] Documentation content changes

## What is the current behavior?

Closes #133 

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: projects/core/components/button/button-options.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import {InjectionToken} from '@angular/core';
+import {TuiAppearance} from '@taiga-ui/core/enums';
+import {TuiSizeXL, TuiSizeXS} from '@taiga-ui/core/types';
+import {PolymorpheusContent} from '@tinkoff/ng-polymorpheus';
+
+export interface ButtonOptions {
+    readonly size: TuiSizeXS | TuiSizeXL;
+    readonly appearance: TuiAppearance | string;
+    readonly shape: 'square' | 'rounded' | null;
+    readonly icon: PolymorpheusContent;
```
Comment (waterplea) [inline projects/core/components/button/button-options.ts:10]: I don't think it makes sense to configure icon with DI. It is different for each instance anyway.

File: projects/core/components/button/button.component.ts [hunk 2]
```diff
@@ -22,11 +22,10 @@ import {
...
 } from '@taiga-ui/cdk';
-import {TuiAppearance} from '@taiga-ui/core/enums';
-import {TuiSizeS, TuiSizeXL, TuiSizeXS} from '@taiga-ui/core/types';
-import {PolymorpheusContent} from '@tinkoff/ng-polymorpheus';
+import {TuiSizeS} from '@taiga-ui/core/types';
 import {Observable} from 'rxjs';
 import {takeUntil} from 'rxjs/operators';
+import {ButtonOptions, TUI_BUTTON_OPTIONS} from './button-options';
 
 @Component({
```
Comment (waterplea) [inline projects/core/components/button/button.component.ts:30]: Let's add interface implementation so the types are more apparent.

File: projects/demo/src/modules/components/button/examples/5/index.html [hunk 3]
```diff
@@ -0,0 +1 @@
+<button tuiButton>Default options</button>
```
Comment (waterplea) [inline projects/demo/src/modules/components/button/examples/5/index.html:1]: Isn't it exactly NOT the default options? 🙂

File: projects/core/components/button/button.component.ts (hunk 2 above)
Comment (vladimirpotekhin) [reply projects/core/components/button/button.component.ts:31]: done

File: projects/demo/src/modules/components/button/examples/5/index.html (hunk 3 above)
Comment (vladimirpotekhin) [reply projects/demo/src/modules/components/button/examples/5/index.html]: u r right) renamed

File: projects/core/components/button/button-options.ts (hunk 1 above)
Comment (vladimirpotekhin) [reply projects/core/components/button/button-options.ts]: removed 👍

**Review by splincode — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

**Pull request merged**